src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1514,23 +1514,62 @@ def __find_disk_from_topology(self, label, pool):
 
         return True
 
-    @accepts()
-    async def unlock_services_restart_choices(self):
+    @accepts(Int('id'))
+    async def unlock_services_restart_choices(self, oid):
         """
         Get a mapping of services identifiers and labels that can be restart
         on volume unlock.
         """
-        svcs = {
+        pool = await self._get_instance(oid)
+
+        services = {
             'afp': 'AFP',
             'cifs': 'SMB',
             'ftp': 'FTP',
             'iscsitarget': 'iSCSI',
             'nfs': 'NFS',
             'webdav': 'WebDAV',
-            'jails': 'Jails/Plugins',
-            'vms': 'Virtual Machines',
         }
-        return svcs
+
+        result = {}
+
+        for k, v in services.items():
+            service = await self.middleware.call('service.query', [['service', '=', k]], {'get': True})
+            if service['enable'] or service['state'] == 'RUNNING':
+                result[k] = v
+
+        try:
+            activated_pool = await self.middleware.call('jail.get_activated_pool')
```

Comment:
If that pool already has jails, it would already be activated and we wouldn't run into this issue, so i guess we can perhaps update this usage to checking if activated pool equals to the current pool name ?